internal/providers/github/common.go

```diff
@@ -167,6 +167,11 @@ func (c *GitHub) ListPackagesByRepository(
 	// create a slice to hold the containers
 	var allContainers []*github.Package
 
+	if c.packageListingClient == nil {
```

Comment:
@jhrozek for my education, in what case would there be no package listing client?

Reply:
> @jhrozek for my education, in what case would there be no package listing client?

I added an explanation to the PR description.